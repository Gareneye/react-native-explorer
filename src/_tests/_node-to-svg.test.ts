import * as yoga from "yoga-layout"

import nodeToSVG from "../node-to-svg"

describe("nodeToSVG", () => {
    it("handles a simple square", () => {
      const rootNode = yoga.Node.create()
      rootNode.setWidth(600)
      rootNode.setHeight(400)
      rootNode.setDisplay(yoga.DISPLAY_FLEX)
      rootNode.setFlexDirection(yoga.FLEX_DIRECTION_ROW)

      const settings = {
        width: 1024,
        height: 768
      }

      const results = nodeToSVG(rootNode, null, settings)
      expect(results).toMatchSnapshot()

      rootNode.free()
    })
})
