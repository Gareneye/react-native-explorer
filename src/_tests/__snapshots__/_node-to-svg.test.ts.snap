// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodeToSVG handles a simple square 1`] = `"<rect x=\\"0\\" y=\\"0\\" width=\\"600\\" height=\\"400\\" fill-opacity=\\"0.1\\" stroke-width=\\"1\\" stroke=\\"black\\"/>"`;
